File: backend/src/main/java/com/example/purchasebackend/domain/enums/PaymentMethod.java
package com.example.purchasebackend.domain.enums;

/**
 * How the user paid for a purchase (the analytics dimension shown in the portal, replacing the old
 * MOCK/GOOGLE_PLAY "billing mode" breakdown).
 *
 * <p>This is independent of {@link BillingMode} (the fulfillment path): in the demo every purchase is
 * fulfilled through MOCK, but the user still picks one of these methods in the SDK popup, and it is
 * recorded per purchase so revenue can be broken down by payment method.
 */
public enum PaymentMethod {
    APPLE_PAY,
    GOOGLE_PLAY,
    PAYPAL,
    CREDIT_CARD
}

File: iap-sdk/src/main/java/com/example/iapsdk/models/PaymentMethod.kt
package com.example.iapsdk.models

/**
 * A payment method the user can choose in the purchase popup. [apiValue] is what the backend records
 * (and breaks revenue down by); [displayName] is what the popup shows.
 */
enum class PaymentMethod(val displayName: String, val apiValue: String) {
    CREDIT_CARD("Credit Card", "CREDIT_CARD"),
    APPLE_PAY("Apple Pay", "APPLE_PAY"),
    GOOGLE_PLAY("Google Pay", "GOOGLE_PLAY"),
    PAYPAL("PayPal", "PAYPAL");

    companion object {
        /** The method selected by default when the popup opens. */
        val DEFAULT = CREDIT_CARD
    }
}
